pkg/apis/migration/v1alpha1/directvolumemigrationprogress_types.go

```diff
@@ -27,20 +27,65 @@ import (
 
 // DirectVolumeMigrationProgressSpec defines the desired state of DirectVolumeMigrationProgress
 type DirectVolumeMigrationProgressSpec struct {
-	ClusterRef *kapi.ObjectReference `json:"clusterRef,omitempty"`
-	PodRef     *kapi.ObjectReference `json:"podRef,omitempty"`
+	ClusterRef   *kapi.ObjectReference `json:"clusterRef,omitempty"`
+	PodRef       *kapi.ObjectReference `json:"podRef,omitempty"`
+	PodSelector  map[string]string     `json:"podSelector,omitempty"`
+	BackOffLimit int                   `json:"backOffLimit,omitempty"`
 }
 
+const (
+	// DVMPDoneLabelKey used to track progress of dvmp operations
+	DVMPDoneLabelKey = "migration.openshift.io/dvmp-done"
+)
+
 // DirectVolumeMigrationProgressStatus defines the observed state of DirectVolumeMigrationProgress
 type DirectVolumeMigrationProgressStatus struct {
-	Conditions                  `json:",omitempty"`
-	PodPhase                    kapi.PodPhase    `json:"phase,omitempty"`
-	ExitCode                    *int32           `json:"exitCode,omitempty"`
-	ContainerElapsedTime        *metav1.Duration `json:"containerElapsedTime,omitempty"`
-	LogMessage                  string           `json:"logMessage,omitempty"`
-	ObservedDigest              string           `json:"observedDigest,omitempty"`
-	LastObservedProgressPercent string           `json:"lastObservedProgressPercent,omitempty"`
-	LastObservedTransferRate    string           `json:"lastObservedTransferRate,omitempty"`
+	Conditions `json:",omitempty"`
+	// RsyncPodStatus observed state of most recent Rsync attempt
+	RsyncPodStatus `json:",inline"`
+	// RsyncPodStatuses history of all Rsync attempts
+	RsyncPodStatuses []RsyncPodStatus `json:"rsyncPodStatuses,omitempty"`
+	// Succeeded whether Rsync operation succeded
+	Succeeded bool `json:"succeeded,omitempty"`
+	// Failed whether Rsync operation failed
+	Failed bool `json:"failed,omitempty"`
+	// Completed wether Rsync operation completed
+	Completed bool `json:"completed,omitempty"`
+	// RsyncElapsedTime total elapsed time of Rsync operation
+	RsyncElapsedTime *metav1.Duration `json:"rsyncElapsedTime,omitempty"`
```

Comment:
Will this be the addition of all the container's duration? or will it be the time elapsed since the start of this CR?

Reply:
@alaypatel07 It will be addition of all container durations

Reply:
can you please update the comment in the previous line to include this?